packages/dashboard/src/components/DashboardFilters/DashboardFiltersEditor/components/FilterEditor.tsx

```diff
@@ -146,80 +166,24 @@ const FilterEditor: React.FC<FilterEditorProps> = ({
     updateFilterProp('key', duplicateLabels.length ? value + ` (${duplicateLabels.length})` : value)
   }
 
-  const isNestedDropDown = filter.filterStyle === FILTER_STYLE.nestedDropdown
+  const isNestedDropdown = filter.filterStyle === FILTER_STYLE.nestedDropdown
 
-  type APIInputProps = {
-    isSubgroup?: boolean
-  }
-  const APIInputs: React.FC<APIInputProps> = ({ isSubgroup = false }) => {
-    const textSelector = isSubgroup ? 'subgroupTextSelector' : 'textSelector'
-    const valueSelector = isSubgroup ? 'subgroupValueSelector' : 'valueSelector'
+  const EditAPIValues = (filter, isNestedDropdown, updateAPIFilter) => {

```

Comment:
Instead of creating a separate component for this can't you just insert the contents of this component into the handleEditAPIValues function call to the overlay?